src/Imdb/Title.php

```diff
@@ -1399,20 +1399,19 @@ public function top250()
     public function plot()
     {
         if (empty($this->plot_plot)) {
-            $page = $this->getPage("Plot");
-            if (empty($page)) {
+            $xpath = $this->getXpathPage("Plot");
+            //$page = $this->getPage("Plot");
+            if (empty($xpath)) {
                 return array();
             } // no such page
-            $doc = new \DOMDocument();
-            @$doc->loadHTML($page);
-            $xp = new \DOMXPath($doc);
-            $cells = $xp->query("//ul[@id=\"plot-summaries-content\"]/li[@id!=\"no-summary-content\"]");
+            $cells = $xpath->query("//ul[@id=\"plot-summaries-content\"]/li[@id!=\"no-summary-content\"]");
             foreach ($cells as $cell) {
                 $link = '';
-                if ($a = $cell->getElementsByTagName('a')->item(0)) {
+                $q = $cell->getElementsByTagName('a')->length - 1;
```

Comment:
What does `$q` stand for? Official variable in the [PHP manual](https://www.php.net/manual/en/domnodelist.item.php) are $index.

Reply:
$q stands for length object.
Should use $length?

Reply:
Okay, you should name your variables to describe what it's being used for. `length object` don't have a `q` in it.

I would prefer it to be `$index` per PHP documentation, but `length` are better than `q` as it grabs the length, but as you change it, it no longer contains the length anymore.

`DOMNodeList::item(int $index):`

Reply:
This may sound stupid?
```
$length = $cell->getElementsByTagName('a')->length;
$useIndex = $length - 1;
```

or should i use this as length `DOMNodeList::item(int $index):`
in that case `$useIndex = $index - 1;`

Reply:
I think all three works, if we stick with general names.

```
$length = $cell->getElementsByTagName('a')->length;
if ($a = $cell->getElementsByTagName('a')->item($length - 1)) {
```

```
$index = $cell->getElementsByTagName('a')->length - 1;
if ($a = $cell->getElementsByTagName('a')->item($index)) {
```

```
$length = $cell->getElementsByTagName('a')->length;
$index = $length - 1;
if ($a = $cell->getElementsByTagName('a')->item($index)) {
```

---

Your variable should however describe what it does. As `$index` can point to anything, it's hard to say what it actually does. `$authorIndex` or `$authorLinkIndex`.

In this example I cache `getElementsByTagName` as we are using it for for length and picking out the item itself. (not tested)

``` php
foreach ($cells as $cell) {
    $link = '';
    $anchors = $cell->getElementsByTagName('a');
    if ($a = $anchors->item($anchors->length - 1)) {
        $href = preg_replace(
            '!/search/title!i',
            'https://' . $this->imdbsite . '/search/title',
            $a->getAttribute('href')
        );
        $link = "\n-\n" . '<a href="' . $href . '">' . trim($a->nodeValue) . '</a>';
    }
    $this->plot_plot[] = $cell->getElementsByTagName('p')->item(0)->nodeValue . $link;
}
```

Reply:
changed to your last option, it is the most meaningful, and represent the actual value

Reply:
@jreklund
Is this PR ready to be approved or are there more changes needed?
Just a question,